bin/your_viewer.py

```diff
@@ -0,0 +1,279 @@
+logging_format = '%(asctime)s - %(funcName)s - %(name)s - %(levelname)s - %(message)s'
+
+# based on https://steemit.com/utopian-io/@hadif66/tutorial-embeding-scipy-matplotlib-with-tkinter-to-work-on-images-in-a-gui-framework
+
+class Paint(Frame):
+    """
+    Class for plotting object
+    """
+
+    # Define settings upon initialization. Here you can specify
+    def __init__(self, master=None):
+
+        # parameters that you want to send through the Frame class.
+        Frame.__init__(self, master)
+
+        #reference to the master widget, which is the tk window
+        self.master = master
+
+        #Creation of init_window
+        #set widget title
+        self.master.title("your_viewer")
+
+        #allowing the widget to take the full space of the root window
+        self.pack(fill=BOTH)#, expand=1)
+        self.create_widgets()
+        #creating a menu instance
+        menu = Menu(self.master)
+        self.master.config(menu=menu)
+
+        #create the file object)
+        file = Menu(menu)
+        
+        #adds a command to the menu option, calling it exit, and the
+        #command it runs on event is client_exit
+        file.add_command(label="Exit", command=self.client_exit)
+
+        #added "file" to our menu
+        menu.add_cascade(label="File", menu=file)
+
+    def create_widgets(self):
+        """
+        Create all the user buttons
+        """
+        #which file to load
+        self.browse = Button(self)
+        self.browse["text"] = "Browse file"
+        self.browse["command"] = self.load_file
+        self.browse.grid(row=0, column=0)
+
+        #move image foward to next gulp of data
+        self.next = Button(self)
+        self.next["text"] = "Next Gulp"
+        self.next["command"] = self.next_gulp
+        self.next.grid(row=0, column=1)
+
+        #move image back to previous gulp of data
+        self.prev = Button(self)
+        self.prev["text"] = "Prevous Gulp"
+        self.prev["command"] = self.prev_gulp
+        self.prev.grid(row=0, column=3)
+
+        #save figure
+        self.prev = Button(self)
+        self.prev["text"] = "Save Fig"
+        self.prev["command"] = self.save_figure
+        self.prev.grid(row=0, column=4)
+        
+    def nice_print(self, dic):
+        """
+        Prints out data files into in a nice to view way
+
+        Inputs:
+        dic --  dictionary containing data file meta data to be printed
+        """
+        for key, item in dic.items():
+            print(f"{key : >27}:\t{item}")
+        
+    def get_header(self):
+        """
+        Gets meta data from data file and give the data to nice_print() to print to user
+        """
+        dic = vars(self.yr.your_header)
+        dic['tsamp'] = self.yr.your_header.tsamp
+        dic['nchans'] = self.yr.your_header.nchans
+        dic['foff'] = self.yr.your_header.foff
+        dic['nspectra'] = self.yr.your_header.nspectra
+        self.nice_print(dic)
+            
+    canvas=''
```

Comment:
`canvas` is sitting outside all functions and does not seem to be doing anything.